### PR TITLE
Removal of Salary Structure Field on Employee Doctype

### PR DESCRIPTION
**Pull request by Talleyrand333**
## Is this a Feature, Chore or Bug?
- [] Feature
- [*] Chore
- [] Bug

## Clearly and concisely describe the feature, chore or bug.
-Removal of Salary Structure Field on Employee Doctype
- Base and Salary Structure made mandatory on Job Offer Doctype

## Analysis and design (optional)
Analyze and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.
- I created a patch to delete the custom field from the database
- I customized the Job Offer form to make the base and salary structure fields mandatory

## Is there a business logic within a doctype?
    - [] Yes
    - [] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.

## Areas affected and ensured
List out the areas affected by your code changes.
- Employee Doctype
- Job Offer Doctype

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.
No , There was no reference found to the deleted field

## Did you test with the following dataset?
- [] Existing Data
- [*] New Data

## Was child table created?
No
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [*] Yes
    - [] No
        If yes, did you write a delete patch?
          Yes

## Is patch required?
- [*] Yes
- [] No
    ## Was the patch test?
    Yes

## Which browser(s) did you use for testing?
  - [*] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/api/doc_events.py [hunk 1]
```diff
@@ -40,16 +39,14 @@ def employee_checkin_validate(doc, method):
 			existing_perm = None
 			checkin_time = get_datetime(doc.time)
 			curr_shift = get_current_shift(doc.employee)
-			last_shift = frappe.get_list("Shift Assignment",fields=["*"],filters={"employee":doc.employee},order_by='creation desc',limit_page_length=1)
-			if curr_shift and last_shift:
+			if curr_shift:
 				start_date = (curr_shift.start_date).strftime("%Y-%m-%d")
 				existing_perm = frappe.db.exists("Shift Permission", {"date": start_date, "employee": doc.employee, "permission_type": perm_map[doc.log_type], "workflow_state": "Approved"})
-				doc.shift_assignment = last_shift[0].name
```
Comment (mymi14s) [inline one_fm/api/doc_events.py:47]: What business has shift assignment in this story?

Comment (Talleyrand333) [reply one_fm/api/doc_events.py:47]: I'm not sure how it happened but It seems other commits are also added. I did not make those changes in my branch.

Comment (mymi14s) [reply one_fm/api/doc_events.py:47]: @Talleyrand333 keep your branches upto date.

Comment (Talleyrand333) [reply one_fm/api/doc_events.py:47]: @mymi14s that seems to be the issue, I did not fetch the latest master-14 branch. I will do that.

**Comment by pjamsheer** [comment]
@Talleyrand333 `Removal of Salary Structure Field on Employee Doctype` need to remove the field from the code base (custom/employee.json(

**Comment by Talleyrand333** [comment]
@pjamsheer I added a patch [here](https://github.com/ONE-F-M/One-FM/blob/183851512/one_fm/patches/v14_0/remove_salary_structure.py). I have also removed the field from the employee.json file.

**Pull request closed without merging**